compiler/rustc_save_analysis/src/dump_visitor.rs

```diff
@@ -1112,7 +1112,7 @@ impl<'tcx> DumpVisitor<'tcx> {
                 name: String::new(),
                 qualname,
                 span,
-                value: filename.to_string(),
+                value: filename.prefer_remapped().to_string(),
```

Comment:
@rust-lang/compiler-contributors & @rust-lang/compiler, who would be the best person to review changes to save-analysis? This PR forces code to explicitly choose if a given use of a source filename should respect the [--remap-path-prefix](https://doc.rust-lang.org/rustc/command-line-arguments.html#--remap-path-prefix-remap-source-names-in-output) commandline option or not. It is unclear to me what save-analysis should do in this case.

Reply:
@Xanewok is a good reviewer I tihnk.

Reply:
This looks correct, we want to have the remapped version here. Thanks for the ping.

Reply:
Thanks, @Xanewok!